superset/datasets/dao.py

```diff
@@ -85,12 +85,23 @@ def validate_table_exists(
             return False
 
     @staticmethod
-    def validate_uniqueness(database_id: int, schema: Optional[str], name: str) -> bool:
```

Comment:
@hughhhh I still think the logic isn't quite write. My understanding is you would like to override an existing dataset—which currently `DatasetDAO.validate_uniqueness(...)` would return  `False` (the name isn't great)—and thus it seems one would just want to bypass the `DatasetDAO.validate_uniqueness(...)` check. 

The problem is I think the `check_count` condition is dependent on the circumstances and thus shouldn't always be `True`. I wonder if the right approach is simply to [pause](https://github.com/sqlalchemy/sqlalchemy/issues/4746) the event listener for specific flows.